sdks/go/pkg/beam/io/filesystem/file.go

```diff
@@ -0,0 +1,44 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filesystem
```

Comment:
I vote it should be part of the `fileio` package, rather than the `filesystem` package. Filesystems, if they're compressed, would be abstracted away when reading from them, while files themselves could be compressed, and it's necessary to know that in order to read them.

I see Python puts it in the filesystem code but it feels tacked on, leading to [repetition for each file system](https://github.com/apache/beam/search?l=Python&q=_get_compression_type), rather than isolating that decision to a general fileio. 

Java keeps it in it's [own class](https://github.com/apache/beam/blob/48bad7d966a583055669850eb9fb558782f636a8/sdks/java/core/src/main/java/org/apache/beam/sdk/io/Compression.java) in the general IOs, and it's largely invoking the CompressedSource or similar delegating to the single implementation around files.